File: jniLib/src/main/java/me/ycdev/android/lib/commonjni/FileStatusHelper.java
package me.ycdev.android.lib.commonjni;

public class FileStatusHelper {
    static {
        CommonJniLoader.load();
    }

    public static class FileStatus {
        public int uid;
        public int gid;
        public int mode;
    }

    public static native FileStatus getFileStatus(String filePath);

}

File: jniLib/src/main/java/me/ycdev/android/lib/commonjni/FileStatusHelper.kt
package me.ycdev.android.lib.commonjni

object FileStatusHelper {
    init {
        CommonJniLoader.load()
    }

    data class FileStatus(var uid: Int = 0, var gid: Int = 0, var mode: Int = 0)

    external fun getFileStatus(filePath: String): FileStatus
}
